Proposal for fleet management overview page (#2185)
giantswarm/docs: src/content/overview/fleet-management/_index.md

  principal:
    parent: overview
    identifier: overview-fleet-management
last_review_date: 2024-05-10
owner:
  - https://github.com/orgs/giantswarm/teams/sig-product
---

> Based on our extensive experience, we understand that orchestrating a large-scale microservices platform poses significant challenges. Ideally, workloads should be designed for seamless execution in the cloud and in containers. However, this becomes exponentially more complex when juggling multiple providers, regions, clusters and environments. Recognizing this, we've developed abstractions to help manage and tackle this complexity effectively.

Reviewer: ```suggestion
Based on our extensive experience, we understand that orchestrating a large-scale microservices platform poses significant challenges. Ideally, workloads should be designed for seamless execution in the cloud and in containers. However, this becomes exponentially more complex when juggling multiple infrastructure providers, regions, clusters and environments. Recognizing this, we've developed abstractions to help manage and tackle this complexity effectively.
```

Reviewer: I think to the public it is not always clear what providers mean

Reviewer: Tried to add a bit more high level fleet management to the intro so people that do not already know what it means get a better understanding of the problem space. WDYT?

Author: Slightly change applied but I like it

> Based on our extensive experience, we understand that orchestrating a large-scale microservices platform poses significant challenges. Ideally, workloads should be designed for seamless execution in the cloud, tailor-made to optimize the container lifecycle. However, the task becomes exponentially more complex when juggling multiple providers, regions, and environments. Recognizing this, we've developed several abstraction layers to help streamline management and tackle this complexity more effectively.

Author: ```suggestion
Based on our extensive experience, we understand that orchestrating a large-scale microservices platform poses significant challenges. Ideally, workloads should be designed for seamless execution in the cloud, tailor-made to optimize the container lifecycle. However, the task becomes exponentially more complex when juggling multiple providers, regions, clusters, and environments. Recognizing this, we've developed several abstraction layers to help streamline management and tackle this complexity more effectively.
```

> ## Features

Author: ```suggestion
## Capabilities
```

Our product is designed to offer a comprehensive set of features to support the construction of robust cloud-native platforms. Key features include:

> - **Multi-environment support**: Every enterprise has unique requirements and constraints, influencing how they design their developer platforms. Platform engineers often struggle to align their organization's structure and processes with the platform while maintaining the system's ability to foster speed and quality, all in compliance with company standards. Our solution, the Giant Swarm platform, simplifies this process. It empowers you to provide developers with various environments, across different regions and providers, flexibly adapted to your organization's needs.

Reviewer: ```suggestion
- **Multi-environment support**: Every enterprise has unique requirements and constraints, influencing how they design their developer platforms. Platform engineers often struggle to align their organization's structure and processes with the platform while maintaining the system's ability to foster speed and quality, all in compliance with company standards. Our solution, the Giant Swarm platform, simplifies this process. It empowers you to provide developers with various environments, across different regions and infrastructures, flexibly adapted to your organization's needs.
```

- **Robust configuration management**: Our bullet-proof configuration management system allows you to manage your clusters, environments, and workloads with flexibility and precision. The system facilitates structured platform configuration with multiple control layers, all from a single repository. It's crafted to adhere to the principles of GitOps, ensuring seamless and efficient management.

> - **Standardized cluster lifecycle management**: Experience a standardized approach to managing your clusters' lifecycle. Giant Swarm leverages the Cluster API implementation, a Kubernetes subproject developed by the open-source community, including Giant Swarm engineers as regular contributors. It allows consistent and reliable cluster lifecycle management, enabling versioning and configuring clusters as code.

Reviewer: ```suggestion
- **Standardized cluster lifecycle management**: Experience a standardized approach to managing your clusters' lifecycle. Giant Swarm leverages the Cluster API implementation, a Kubernetes subproject developed by the open-source community, including Giant Swarm engineers as regular contributors. What Giant Swarm adds on top, are enterprise-grade features and configuration out-of-the-box as well as a battle tested versioned package including all base functionality a production-ready cluster needs. This allows for consistent and reliable cluster lifecycle management, enabling versioning and configuring clusters as code.
```

tried to add a bit of what we add on top

## Cloud-native applications

The platform is heavily relying on [Cluster API implementation]({{< relref "/overview/fleet-management/introduction_cluster-api" >}}) to manage the lifecycle of your clusters. It opens up the possibility to manage your clusters in many providers adapting to the infrastructure nuances of each solution while keeping a consistent experience for the user.

> Every company requires a different level of control over their infrastructure and its configuration. We work with upstream to offer you a platform that is compliant with all the standards and processes your company imposes.

Reviewer 2: We need to make explicit points in here. This is too vague and not saying more than the bullet list above.

  principal:
    parent: overview
    identifier: overview-fleet-management

> last_review_date: 2024-05-10

Author: ```suggestion
last_review_date: 2024-05-27
```
